src/webportal/src/app/components/util/job.js

```diff
@@ -94,3 +96,43 @@ export function getJobModifiedTimeString(job) {
     return 'N/A';
   }
 }
+
+export function isLongRunJob(job) {
+  return (Date.now() - job.createdTime) > MIN_ABNORMAL_JOB_DURATION_MILLISECOND;
+}
+
+export function isLowGpuUsageJob(job) {
+  return !isNil(job.gpuUsage) && Number(job.gpuUsage) < 10;
+}
+
+export function listAbnormalJobs(allJobs, lowGpuJobsInfo) {
+  const allRuuingJobs = allJobs.filter((job) => job.state === 'RUNNING');
+  const longRunJobs = allRuuingJobs.filter(isLongRunJob);
+
+  // Get low GPU usage jobs
+  const lowGpuUsageJobs = allRuuingJobs.reduce((acc, cur)=>{
+    const gpuUsageInfo = lowGpuJobsInfo.find((info) => info.jobName === cur.name);
+    if (isNil(gpuUsageInfo)) {
+      return acc;
+    }
+    const lowGpuUsageJob = {...cur};
+    lowGpuUsageJob['gpuUsage'] = gpuUsageInfo.gpuUsage;
+    acc.push(lowGpuUsageJob);
+    return acc;
+  }, []);
+
+  // Merge long run jobs and low GPU usage jobs
+  const abnormalJobs = [...longRunJobs];
```

Comment:
A shallow copy is created here. The original data will be edited.
I'm not sure if it is by design.

Reply:
It's a mistake, thanks